### PR TITLE
Feature/hyp 56 swagger UI appearance api

### PR DESCRIPTION
**Pull request by dc-andysign**
---
name: Swagger UI appearance for Hyproof API
about: Feature/hyp 56 swagger customisable UI appearance for Hyproof API
---

## Summary

IMAGE

This PR follows the requirements described in HYP-56.

This change is designed to allow the user to customise the styling for the Swagger OpenAPI Documentation GUI and that can be done with the use of ENV variables.

## Motivation

When demonstrating this project or when developing it is easy to get confused and forget which Swagger API GUI is which, therefore when using multiple GUIs the appearance need to change.

## Describe alternatives you've considered

There might be other ways of doing this task but this seems to be good enough.

## Additional context

With the help of the **`customCss`** option, the can be changed as in the background. The HTML page title can be changed with the help of the **`customSiteTitle`** option. Last but not least, the heading H2 can be changed with the **`swaggerJson.info.title`** option.

E.g:

```js
  const swaggerJson.info.title += `:${API_SWAGGER_HEADING}`

  const options: SwaggerUiOptions = {
  	swaggerOptions: "..",
  	customCss: "...",
  	customSiteTitle: "....",
  }

  app.use('/swagger', serve, setup(undefined, options))
```

---

### CODE REVIEW

File: src/controllers/health/__tests__/index.test.ts
Comment (rmlearney-digicatapult) [inline src/controllers/health/__tests__/index.test.ts:1]: Why is a health test being pulled into this PR?

File: src/controllers/v1/transaction/index.ts
Comment (rmlearney-digicatapult) [inline src/controllers/v1/transaction/index.ts:1]: Do you need to rebase?

Comment (dc-andysign) [reply src/controllers/v1/transaction/index.ts:1]: Not anymore .. 

This PR only includes **`src/env.ts`** , **`README.md`** and **`package*`** :)

File: README.md [hunk 1]
```diff
@@ -136,7 +139,7 @@ docker-compose -f docker-compose-3-personal.yml logs -f | grep regulator
 ```
 
 > single persona
-> Run `docker composel up -d` to start the required dependencies to fully demo `dscp-hyproof-api`.
+> Run `docker compose up -d` to start the required dependencies to fully demo `dscp-hyproof-api`.
 
 > multiple persona
 > Run `docker-compose -f docker-compose-3-personal.yml up` to start the required dependencies to fully demo `dscp-hyproof-api`.
```
[the remark this answers is not in the document]
Comment (dc-andysign) [reply README.md:145]: Doing it now .. give me 10 ~15 min

[the remark this answers is not in the document]
Comment (dc-andysign) [reply README.md]: Fixed

File: docker-compose-3-persona.yml [hunk 2]
```diff
@@ -135,6 +135,9 @@ services:
       - IPFS_HOST=ipfs
       - IPFS_PORT=5001
       - USER_URI=//Alice
+      - API_SWAGGER_BG_COLOR=#e4f2f3
+      - API_SWAGGER_TITLE=API-Heidi
```
Comment (rmlearney-digicatapult) [inline docker-compose-3-persona.yml:139]: Do we need 'API-xxx' here, or just the names?

File: package.json [hunk 3]
```diff
@@ -10,6 +10,9 @@
     "build:tsoa": "tsoa spec-and-routes",
     "build": "tsoa spec-and-routes && tsc",
     "start": "node build/index.js",
+    "start:persona:01": "API_SWAGGER_BG_COLOR='#e4f2f3' API_SWAGGER_TITLE=API-Heidi    API_SWAGGER_HEADING=API-Heidi     node build/index.js",
```
Comment (rmlearney-digicatapult) [inline package.json:13]: Do we need 'API-xxx' here or just the names 

i.e. 'Heidi' vs. 'API-Heidi' - I don't think the 'API-' adds anything but it's not that important

**Comment by rmlearney-digicatapult** [comment]
Tested & works

File: docker-compose-3-persona.yml (hunk 2 above)
Comment (dc-andysign) [reply docker-compose-3-persona.yml:139]: Just the name but keep in mind , the dev doing the demo will have another tab open for identity and we need to tell them apart ..

File: package.json (hunk 3 above)
Comment (dc-andysign) [reply package.json:13]: Up to you .. i can change it ..

File: docker-compose-3-persona.yml [hunk 4]
```diff
@@ -188,16 +194,19 @@ services:
...
+      - 'DB_HOST=energy-owner-postgres-identity'
+      - 'DB_PORT=5432'
+      - 'DB_NAME=dscp-identity'
+      - 'DB_USERNAME=postgres'
+      - 'DB_PASSWORD=postgres'
+      - 'SELF_ADDRESS=5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty'
+      - 'AUTH_TYPE=${AUTH_TYPE:-NONE}'
+      - 'API_SWAGGER_BG_COLOR=#e7f6e6'
+      - 'API_SWAGGER_TITLE=IdentityAPIemma'
+      - 'API_SWAGGER_HEADING=Identity-API-System-emma'
```
Comment (mattdean-digicatapult) [inline docker-compose-3-persona.yml:209]: Seems like a funny naming/casing convension. Is there a reason we didn't go with spaces in the end other than you don't like the look?

Also I think we should get the word `HyProof` in there somewhere, so maybe something like `Hyproof Indentity API - Emma`

File: docker-compose-3-persona.yml [hunk 5]
```diff
@@ -121,20 +124,23 @@ services:
...
+      - 'DB_NAME=dscp-hyproof-api'
+      - 'DB_USERNAME=postgres'
+      - 'DB_PASSWORD=postgres'
+      - 'IDENTITY_SERVICE_HOST=hydrogen-producer-identity'
+      - 'IDENTITY_SERVICE_PORT=9000'
+      - 'IPFS_HOST=ipfs'
+      - 'IPFS_PORT=5001'
+      - 'USER_URI=//Alice'
+      - 'API_SWAGGER_BG_COLOR=#e4f2f3'
+      - 'API_SWAGGER_TITLE=API-Heidi'
```
Comment (mattdean-digicatapult) [inline docker-compose-3-persona.yml:142]: Again I think we should get the word `HyProof` in here so maybe something like `Hyproof API - Heidi`

File: docker-compose-3-persona.yml [hunk 6]
```diff
@@ -47,27 +47,27 @@ services:
...
-    #      - 5432:5432
+    # ports:
+    #   - 5432:5432
     volumes:
       - hydrogen-producer-hyproof-api-storage:/var/lib/postgresql/data
     environment:
-      - POSTGRES_PASSWORD=postgres
-      - POSTGRES_USER=postgres
-      - POSTGRES_DB=dscp-hyproof-api
+      - 'POSTGRES_PASSWORD=postgres'
```
Comment (mattdean-digicatapult) [inline docker-compose-3-persona.yml:55]: is there a reason for all the diff churn adding the `'` wrappings?

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
Implementation looks great, I think we just need some changes to copy and perhaps to undo some of the diff churn

File: docker-compose-3-persona.yml (hunk 6 above)
Comment (dc-andysign) [reply docker-compose-3-persona.yml:55]: Hmmm.. i guess it is easier when every element is wrapped with quotes .. if something ends with `#fff` it is easier to tell `# something` is not a comment if you have `V=#fff` but i can put it back , no problem .. just tell me

File: docker-compose-3-persona.yml [hunk 7]
```diff
@@ -83,16 +83,19 @@ services:
...
-      - API_HOST=hydrogen-producer-node
-      - API_PORT=9944
-      - DB_HOST=hydrogen-producer-postgres-identity
-      - DB_PORT=5432
-      - DB_NAME=dscp-identity
-      - DB_USERNAME=postgres
-      - DB_PASSWORD=postgres
-      - SELF_ADDRESS=5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
-      - AUTH_TYPE=${AUTH_TYPE:-NONE}
+      - 'PORT=9000'
```
Comment (rmlearney-digicatapult) [inline docker-compose-3-persona.yml:86]: Why do these all need wrapping with ' They worked fine before?

Only add to the new API_SWAGGER stuff if necessary for escaping spaces

Comment (dc-andysign) [reply docker-compose-3-persona.yml:86]: On it ..

File: docker-compose-3-persona.yml (hunk 6 above)
Comment (mattdean-digicatapult) [reply docker-compose-3-persona.yml]: the main issue here is the diff churn and as this is just an escaping syntax it's something to just know if using yaml anyway

File: docker-compose-3-persona.yml [hunk 8]
```diff
@@ -198,6 +204,9 @@ services:
       - DB_PASSWORD=postgres
       - SELF_ADDRESS=5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty
       - AUTH_TYPE=${AUTH_TYPE:-NONE}
+      - 'API_SWAGGER_BG_COLOR=#e7f6e6'
+      - 'API_SWAGGER_TITLE=Hyproof Indentity API - Emma'
```
Comment (mattdean-digicatapult) [inline docker-compose-3-persona.yml:208]: Sorry typo in my suggestion. `Identity`

File: docker-compose-3-persona.yml [hunk 9]
```diff
@@ -93,6 +93,9 @@ services:
       - DB_PASSWORD=postgres
       - SELF_ADDRESS=5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
       - AUTH_TYPE=${AUTH_TYPE:-NONE}
+      - 'API_SWAGGER_BG_COLOR=#e4f2f3'
+      - 'API_SWAGGER_TITLE=Hyproof Indentity API - Heidi'
```
Comment (rmlearney-digicatapult) [inline docker-compose-3-persona.yml:97]: Identity rather than Indentity

File: docker-compose-3-persona.yml [hunk 10]
```diff
@@ -93,6 +93,9 @@ services:
...
+      - 'API_SWAGGER_HEADING=Hyproof Indentity API - Heidi'
```
Comment (rmlearney-digicatapult) [inline docker-compose-3-persona.yml:98]: HyProof rather than Hyproof

**Review by mattdean-digicatapult — APPROVED** [review]
🚀

**Comment by rmlearney-digicatapult** [comment]
Looks good. 

But no love for camel case HyProof? :(

**Review by mattdean-digicatapult — DISMISSED** [review]

**Review by n3op2 — APPROVED** [review]

**Pull request merged**